### PR TITLE
Add `Waiting for: Product Owner` across all issues where a community member comments

### PR DESCRIPTION
**Pull request by hubertdeng123**
see title

### CODE REVIEW

**Comment by hubertdeng123** [comment]
This should not include PR's. Maybe in the future, but there are some gaps right now that need to be addressed before that happens. Primarily, PR's do not have Product Areas, so slack notifications will break.

File: src/brain/issueLabelHandler/followups.ts [hunk 1]
```diff
@@ -44,11 +44,9 @@ function isContractor(payload) {
   return payload.comment.author_association === 'COLLABORATOR';
 }
 
-function isNotWaitingForCommunity(payload) {
+function isPullRequest(payload) {
   const { issue } = payload;
-  return !issue?.labels.some(
-    ({ name }) => name === WAITING_FOR_COMMUNITY_LABEL
-  );
+  return issue.pull_request ? true : false;
```
Comment (azaslavsky) [inline src/brain/issueLabelHandler/followups.ts:49]:
```suggestion
  return !!issue.pull_request;
```

File: src/brain/issueLabelHandler/index.test.ts [hunk 2]
```diff
@@ -159,6 +168,7 @@ describe('issueLabelHandler', function () {
       {
         ...makePayload(repo, undefined, username),
         comment: { author_association: membership },
+        ...(isPR ? { issue: { pull_request: {} } } : {}),
```
Comment (azaslavsky) [inline src/brain/issueLabelHandler/index.test.ts:171]: nit: I *think* I get what you're doing here ("only if`isPR === true` add the field `issue: { pull_request: {} }` to the object I am building"), but its pretty hard to read in practice.

**Review by azaslavsky — APPROVED** [review]

**Comment by hubertdeng123** [comment]
This may be annoying when a product owner removes the `Waiting for: Product Owner` label and wants the item backlogged. Then multiple people comment on the issue and automation adds back `Waiting for: Product Owner`. 🤔

**Review by chadwhitacre — APPROVED** [review]

**Comment by chadwhitacre** [comment]
> This may be annoying when a product owner removes the `Waiting for: Product Owner` label and wants the item backlogged. Then multiple people comment on the issue and automation adds back `Waiting for: Product Owner`. 🤔

Ftr we had a conversation about this and decided to proceed as-is for now. If there's an active conversation going on (like every few minutes back and forth) the product owner can leave "Waiting for" on there until the conversation dies down. What happens when a user comments and "Waiting for: Product Owner" is already applied? That's a no-op, correct?

**Comment by hubertdeng123** [comment]
> What happens when a user comments and "Waiting for: Product Owner" is already applied? That's a no-op, correct?

Correct.

**Pull request merged**